yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -207,18 +211,20 @@ class FilterOperator : public ObservableOperator<U, U> {
       : ObservableOperator<U, U>(std::move(upstream)),
         function_(std::move(function)) {}
 
-  void subscribe(Reference<Observer<U>> observer) override {
+  Reference<Subscription> subscribe(Reference<Observer<U>> observer) override {
+    auto subscription = make_ref<FilterSubscription>(get_ref(this), std::move(observer));
     ObservableOperator<U, U>::upstream_->subscribe(
```

Comment:
ditto

Reply:
yeah, the above comment applies to all operators. Please note that all the operator subscription class (eq. FilterSubscription) is implementing both Subscriber and Subscription which needs to be properly chained to the upstream subscribers. 
(its easy to get lost in the complexity when debugging)